### PR TITLE
OJ-2320: added alarms for the dead Letter event queue and the audit e…

### PR DESCRIPTION
**Pull request by liamStockwellGDS**
…vent queue

NINO | Platform | Add alarms for AuditQueue and Dead Letter Queue

Proposed changes
What changed
added alarms to template file.

Why did it change
To allow for alarms

Issue tracking
[OJ-2320](https://govukverify.atlassian.net/browse/OJ-2320)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[OJ-2320]: https://govukverify.atlassian.net/browse/OJ-2320?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: txma/template.yaml [hunk 1]
```diff
@@ -224,3 +224,53 @@ Outputs:
     Value: !GetAtt AuditEventDeadLetterQueue.Arn
     Export:
       Name: !Sub ${AWS::StackName}-AuditEventDLQArn
+
+  AuditEventSQSOldMessagesAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Condition: "DeployAlarms"
+    Properties:
+      AlarmName: !Sub "${AWS::StackName}-AuditEventSQSOldMessagesAlarm"
+      AlarmDescription: !Sub "Trigger an alarm when the age of the oldest audit event message is 5 or more minutes. ${SupportManualURL}"
```
Comment (surajkumar) [inline txma/template.yaml:233]: Maybe rename this to be more clear e.g.  "Alarm for when a message has stayed on the queue for more than 5 minutes", similarly with the resource name

File: txma/template.yaml [hunk 2]
```diff
@@ -224,3 +224,53 @@ Outputs:
     Value: !GetAtt AuditEventDeadLetterQueue.Arn
     Export:
       Name: !Sub ${AWS::StackName}-AuditEventDLQArn
+
+  AuditEventSQSOldMessagesAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Condition: "DeployAlarms"
```
Comment (surajkumar) [inline txma/template.yaml:230]: This condition does not exist so the template will fail to deploy. Could you please add the condition, remove the `"` and make sure you have tested the alarm.

File: txma/template.yaml (hunk 1 above)
Comment (Caitlin-cooling) [inline txma/template.yaml:233]: Where is this `SupportManualURL` defined?

File: txma/template.yaml [hunk 3]
```diff
@@ -224,3 +224,53 @@ Outputs:
...
+    Properties:
+      AlarmName: !Sub "${AWS::StackName}-AuditEventSQSOldMessagesAlarm"
+      AlarmDescription: !Sub "Trigger an alarm when the age of the oldest audit event message is 5 or more minutes. ${SupportManualURL}"
+      ActionsEnabled: true
+      AlarmActions:
+        - !ImportValue platform-alarm-topic-slack-warning-alert
```
Comment (Caitlin-cooling) [inline txma/template.yaml:236]: This value looks a bit different to the AlarmActions we have in NINo - is there a reason for this?

File: txma/template.yaml [hunk 4]
```diff
@@ -224,3 +224,53 @@ Outputs:
...
+        - Name: QueueName
+          Value: !GetAtt AuditEventQueue.QueueName
+      Period: 60
+      EvaluationPeriods: 1
+      DatapointsToAlarm: 1
+      Threshold: 300
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      TreatMissingData: notBreaching
+
+  AuditEventDLQMessagesVisibleAlarm:
```
Comment (surajkumar) [inline txma/template.yaml:253]: Could anything from the answers on SO be helpful here? https://stackoverflow.com/questions/60211243/configure-sqs-dead-letter-queue-to-raise-a-cloud-watch-alarm-on-receiving-a-mess

**Review by surajkumar — CHANGES_REQUESTED** [review]

**Review by surajkumar — CHANGES_REQUESTED** [review]

File: txma/template.yaml [hunk 5]
```diff
@@ -5,18 +5,38 @@ Transform: AWS::LanguageExtensions
 Parameters:
   PermissionsBoundary:
     Type: String
-    Default: ""
+    Default: "none"
```
Comment (Caitlin-cooling) [inline txma/template.yaml:8]: Think we can revert this back now

File: txma/template.yaml [hunk 6]
```diff
@@ -5,18 +5,38 @@ Transform: AWS::LanguageExtensions
...
   Environment:
     Description: The target environment
     Type: String
     Default: dev
     AllowedValues: [dev, build, staging, integration, production]
     ConstraintDescription: must be dev, build, staging, integration or production
+  TxmaStackName:
```
Comment (Caitlin-cooling) [inline txma/template.yaml:15]: Sorry what do we need this for? We're in the txma stack no? 🤔

File: txma/template.yaml [hunk 7]
```diff
@@ -72,7 +92,6 @@ Resources:
 
   AuditEventDeadLetterQueue:
     Type: AWS::SQS::Queue
-    Condition: IsNotDevEnvironment
```
Comment (Caitlin-cooling) [inline txma/template.yaml:75]: Think @CharlesIC put these in for a reason - is there a reason you're taking them out?

File: txma/template.yaml [hunk 8]
```diff
@@ -192,7 +287,7 @@ Outputs:
     Description: The name of the SQS audit event queue. Needed to give write permissions to lambdas
     Value: !GetAtt AuditEventQueue.QueueName
     Export:
-      Name: AuditEventQueueName
+      Name:  !Sub ${AWS::StackName}-AuditEventQueueNameToRemove
```
Comment (Caitlin-cooling) [inline txma/template.yaml:290]: I know this is needed for local dev but we are sure that nothing is importing using the old name?

**Review by CharlesIC — APPROVED** [review]

File: txma/template.yaml [hunk 9]
```diff
@@ -223,4 +313,4 @@ Outputs:
     Description: The ARN of the audit event SQS queue DLQ
     Value: !GetAtt AuditEventDeadLetterQueue.Arn
     Export:
-      Name: !Sub ${AWS::StackName}-AuditEventDLQArn
+      Name: !Sub ${AWS::StackName}-AuditEventDLQArn
```
Comment (CharlesIC) [inline txma/template.yaml:316]:
```suggestion
      Name: !Sub ${AWS::StackName}-AuditEventDLQArn
      
```
Restore the newline at the end of the file

**Review by surajkumar — APPROVED** [review]

**Pull request merged**